### ISSUE TITLE
[BUG] `npm owner` does not support workspaces

### ISSUE THREAD

**Issue by nlf** [npm/cli#4600]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

running `npm owner` commands from within a workspace in a project logs the following error

```
npm ERR! code ENOWORKSPACES
npm ERR! This command does not support workspaces.
```

### Expected Behavior

commands should work as expected. we'll need to modify the `getPkg` method such that it is workspace aware, as well as add the `execWorkspaces` method which appears it can call `exec` directly.

### Steps To Reproduce

1. Clone this repo
2. `cd workspaces/arborist`
3. `npm owner ls`
4. See error...

### Environment

- npm: 8.5.5
- Node.js: 16.13.2
- OS Name: OSX
- System Model Name: M1 MBP
- npm config:
```ini
; copy and paste output from `npm config ls` here
```

**Comment by darcyclarke** [comment]
Interesting, looks like `npm doctor` also is throwing in this setting. I know there was an initial set of commands we decided wouldn't support workspaces out the gate (ref. https://github.com/npm/rfcs/blob/main/accepted/0038-workspaces-run-cmds.md#3-unsupported) but given that we now support auto detection, I think we need to just triple check each command works when navigating into a workspace dir.

**Comment by fritzy** [comment]
@darcyclarke you may want to move this to a priority 1, since this is due to recent changes.

**Comment by alexjoverm** [comment]
Agree with @fritzy here. We use semantic-release at Storyblok to automatically release all our SDK packages. Semantic release runs `npm owner` under the hood, and due to this issue, the releases of our whole SDK ecosystem were broken.

In case this is happening to you, for us downgrading to `16.13.2` on our Github Actions config worked well 👌 https://github.com/storyblok/storyblok-js/runs/5855290031

**Comment by bnb** [comment]
> Interesting, looks like npm doctor also is throwing in this setting. I know there was an initial set of commands we decided wouldn't support workspaces out the gate (ref. https://github.com/npm/rfcs/blob/main/accepted/0038-workspaces-run-cmds.md#3-unsupported) but given that we now support auto detection, I think we need to just triple check each command works when navigating into a workspace dir.

I just ran into `npm doctor` not supporting workspaces. Hilariously, I tried running it to see if it'd be able to help me figure out why one of my workspaces isn't working despite all my configuration being *apparently* correct.

**Comment by wraithgar** [comment]
[This is fixed in v8.9.0](https://github.com/npm/cli/blob/latest/CHANGELOG.md#v890-2022-05-04)

**Comment by mattclough1** [comment]
Is it expected that `npm set` gives the same error?

**Edit:** I see this is covered in #6099
